common/transport/amqp/src/sender_link.ts

```diff
@@ -314,6 +315,22 @@ export class SenderLink extends EventEmitter implements AmqpLink {
             //
             let error = this._indicatedError; // This could be undefined.
             this._indicatedError = undefined;
+            if (!error) {
+              ///
+              // question for reviewers: Is this a Rhea bug?
+              //
+              // https://github.com/amqp/rhea/blob/e826e8bbffd410eaa5a1efc70650f37c14069b23/lib/link.js#L161-L163
+              //
+              // when they fire the link_error and link_close events, I think context.error should be set.
+              // Since it's not set, we have to guess that there's an error by looking at link state here.
+              //
+              // I think my change is OK?  Do you agree?  Should we push for a Rhea fix?
+              //
+              const senderState = (context.sender as any).state;
```

Comment:
is context.sender unknown? why are we coercing it to any?

Reply:
context.sender is of type `Sender` from `link.d.ts` which doesn't have a state property defined.  I guess that rhea bug, if we open it, could also update the typing to add `Sender.state`.

Reply:
is there a problem with changing this line to `(context.sender as unknown).state`?

Reply:
I don't want to coerce this to `unkonwn`.  That would cause a type error.  

`any` is most permissive (meaning we can call any methods and access any properties on it).  `unkonwn` is least permissive (meaning we can't call any methods on it or access any properties on it).  `unknown` is used when we have an object, and we don't konw the type, and we don't care what the type is because we're just holding it or passing it through. 

[https://devblogs.microsoft.com/typescript/announcing-typescript-3-0-rc-2/#the-unknown-type](https://devblogs.microsoft.com/typescript/announcing-typescript-3-0-rc-2/#the-unknown-type)